### ISSUE TITLE
Exposing game servers via service

### ISSUE THREAD

**Issue by hyperbolic2346** [DrPsychick/ark-server-charts#13]
I'm getting back to running an ark server on my bare metal k8s cluster and I use metallb to expose loadbalancer IPs that route directly to the server in my past implementation without helm. Is this something you'd be interested in including in this helm chart? I don't see how to handle this with nodeports unless I manually add the route and update it as the server hops around nodes. I don't want to pin it to a specific node.

**Comment by hyperbolic2346** [comment]
You can actually have it set to only advertise the IP on the machines currently running the workload.

So you set up a service as load balancer and then metallb assigns an IP and uses either ARP or BGP to route traffic. By default it advertises all nodes and traffic is routed normally through your calico/flannel, etc. Or you can have metallb only advertise the IP on the node(s) with the workload.

I use this extensively to poke a hole in my firewall and have the traffic land on the correct node for the work. It isn't a direct flow, but it is minimal hops and acceptable to me.

**Comment by Dracozny** [comment]
This is the method I use. I actually use metallb in bgp mode but it seemed to work in L2 mode as well.
in this case it targets any pod in the deployment. Since we are only launching 1 pod at any given time for the deployment it seems to work without issue. 

```
cat mlb-extinction_mono.yaml
apiVersion: v1
kind: Service
metadata:
  name: ark-extinction-svc
  namespace: ark
spec:
  type: LoadBalancer
  loadBalancerIP: 192.168.3.50
  selector:
     app.kubernetes.io/component: extinction
  ports:
  - name: rcon
    protocol: TCP
    port: 32330
    targetPort: 32330
  - name: gclient
    protocol: UDP
    port: 7777
    targetPort: 7777
  - name: gclient-p
    protocol: UDP
    port: 7778
    targetPort: 7778
  - name: server-list
    protocol: UDP
    port: 27015
    targetPort: 27015
```

**Comment by DrPsychick** [comment]
solved with #29 ?
